packages/react-native-gesture-handler/src/v3/hooks/events/useTouchEvent.ts

```diff
@@ -0,0 +1,65 @@
+import { GestureTouchEvent } from '../../../handlers/gestureHandlerCommon';
+import {
+  compareTags,
+  runWorkletCallback,
+  touchEventTypeToCallbackType,
+} from '../utils';
+import { Reanimated } from '../../../handlers/gestures/reanimatedWrapper';
+import { TouchEventType } from '../../../TouchEventType';
+import { CallbackHandlers, TouchEvent } from '../../types';
+import { NativeSyntheticEvent } from 'react-native';
+
+export function useTouchEvent(
+  handlerTag: number,
+  config: any,
+  shouldUseReanimated: boolean
+) {
+  const handlers: CallbackHandlers = {
+    onTouchesDown: config.onTouchesDown,
+    onTouchesMove: config.onTouchesMove,
+    onTouchesUp: config.onTouchesUp,
+    onTouchesCancelled: config.onTouchesCancelled,
+  };
+
+  const onGestureHandlerTouchEvent = (event: TouchEvent) => {
+    'worklet';
+
+    if (!compareTags(handlerTag, event)) {
+      return;
+    }
+
+    if (
+      // @ts-ignore That's the point, we want to check if nativeEvent exists or not
+      event.nativeEvent &&
+      (event as NativeSyntheticEvent<GestureTouchEvent>).nativeEvent
+        .eventType !== TouchEventType.UNDETERMINED
+    ) {
+      runWorkletCallback(
+        touchEventTypeToCallbackType(
+          (event as NativeSyntheticEvent<GestureTouchEvent>).nativeEvent
+            .eventType
+        ),
+        handlers,
+        event
+      );
```

Comment:
```suggestion
    if (
      isNativeEvent(event) &&
      event.nativeEvent.eventType !== TouchEventType.UNDETERMINED
    ) {
      runWorkletCallback(
        touchEventTypeToCallbackType(event.nativeEvent.eventType),
        handlers,
        event
      );
```

We can remove some of these casts and the `@ts-ignore` if we use `isNativeEvent`.